### PR TITLE
Filter exported point clouds by polygon layer

### PR DESCRIPTION
**Pull request by uclaros**
## Description
This is a #49280 followup, adding support for spatially filtering the exported point cloud layer by a polygon layer.
Also allows exporting to uncompressed LAS files.

image

Funded by: State of Vorarlberg and Swiss QGIS user group

### CODE REVIEW

**Review by wonder-sk — APPROVED** [review]

File: src/core/pointcloud/qgspointcloudlayerexporter.h [hunk 1]
```diff
@@ -45,6 +45,51 @@ class CORE_EXPORT QgsPointCloudLayerExporter SIP_NODEFAULTCTORS
...
+
+    /**
+     * Gets the extensions filter for the specified \a format
+     */
+    static QString getFilter( ExportFormat format );
+
+    /**
+     * Gets the OGR driver name for the specified \a format
+     */
+    static QString getOgrDriverName( ExportFormat format );
```
Comment (wonder-sk) [inline src/core/pointcloud/qgspointcloudlayerexporter.h:91]: I would move this one out of the public API - it feels like implementation detail...

File: src/core/pointcloud/qgspointcloudlayerexporter.h [hunk 2]
```diff
@@ -227,7 +265,7 @@ class CORE_EXPORT QgsPointCloudLayerExporter SIP_NODEFAULTCTORS
 
     QgsPointCloudIndex *mIndex = nullptr;
     QString mName = QObject::tr( "Exported" );
-    QString mFormat;
+    ExportFormat mFormat;
```
Comment (wonder-sk) [inline src/core/pointcloud/qgspointcloudlayerexporter.h:268]: It seems the format value stays uninitialized - would be good to set some valid initial value

File: src/core/pointcloud/qgspointcloudlayerexporter.h [hunk 3]
```diff
@@ -45,6 +45,51 @@ class CORE_EXPORT QgsPointCloudLayerExporter SIP_NODEFAULTCTORS
...
+
+    /**
+     * Gets the translated name for the specified \a format
+     */
+    static QString getTranslatedName( ExportFormat format );
+
+    /**
+     * Gets the extensions filter for the specified \a format
+     */
+    static QString getFilter( ExportFormat format );
```
[the remark this answers is not in the document]
Comment (uclaros) [reply src/core/pointcloud/qgspointcloudlayerexporter.h]: Indeed, that's gui stuff!

File: src/core/pointcloud/qgspointcloudlayerexporter.h (hunk 1 above)
Comment (uclaros) [reply src/core/pointcloud/qgspointcloudlayerexporter.h]: SIP_SKIPped!

**Review by wonder-sk — APPROVED** [review]

**Pull request merged**